integration-tests/azure-cosmos/src/main/java/io/quarkiverse/azure/cosmos/it/CosmosResource.java

```diff
@@ -0,0 +1,82 @@
+package io.quarkiverse.azure.cosmos.it;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.*;
+
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.CosmosDatabase;
+import com.azure.cosmos.models.*;
+
+@Path("/quarkus-azure-cosmos")
+@ApplicationScoped
+public class CosmosResource {
+
+    @Inject
+    CosmosClient cosmosClient;
+
+    @Path("/{database}/{container}")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createItem(
+            @PathParam("database") String database,
+            @PathParam("container") String container,
+            Item body,
+            @Context UriInfo uriInfo) {
+
+        getContainer(database, container).upsertItem(body);
+        return Response.created(uriInfo.getAbsolutePathBuilder().path(body.getId()).build()).build();
+    }
+
+    @Path("/{database}/{container}/{itemId}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getItem(
+            @PathParam("database") String database,
+            @PathParam("container") String container,
+            @PathParam("itemId") String itemId) {
+        CosmosItemResponse<Item> item = getContainer(database, container).readItem(itemId, new PartitionKey(itemId),
+                Item.class);
+
+        return Response.ok().entity(item.getItem()).build();
+
+    }
+
+    @Path("/{database}/{container}/{itemId}")
+    @DELETE
+    public Response deleteItem(
+            @PathParam("database") String database,
+            @PathParam("container") String container,
+            @PathParam("itemId") String itemId) {
+        getContainer(database, container).deleteItem(itemId, new PartitionKey(itemId), new CosmosItemRequestOptions());
+
+        return Response.noContent().build();
```

Comment:
the getContainer() will create the database or container if it doesn't exist.
so what if someone call `deleteItem` with `/notexitdb1/notexistcontainer2/notexititem3`
it will create a new db and a new container?

Reply:
`However, you cannot use any Azure Cosmos DB data plane SDK to authenticate management operations with a Microsoft Entra identity, so you need to create database and container manually.`
As you said, what if users call the method with a Microsoft Entra identity?

Reply:
It will fail with the similar error message `The given request [POST /dbs] cannot be authorized by AAD token in data plane. Learn more: https://aka.ms/cosmos-native-rbac.`

Reply:
> the getContainer() will create the database or container if it doesn't exist. so what if someone call `deleteItem` with `/notexitdb1/notexistcontainer2/notexititem3` it will create a new db and a new container?

Good catch. The code should only create database /  container if user tries to create/update a new/existing item. Refined in 29a9fd7fe167b169fb7481880db5896b00c8d33f.